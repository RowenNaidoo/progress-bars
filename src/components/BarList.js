import React from 'react'
import PropTypes from 'prop-types'
import Bar from './Bar'
import styled from 'styled-components'

const BarList = ({ bars }) => (
  bars
    .map((val, index) => <BarContainer key={`Bar_${index}`}><Bar barValue={val} /></BarContainer>)
)

const BarContainer = styled.div`
  margin: 10px 0;
`

BarList.propTypes = {
  bars: PropTypes.arrayOf(Number)
}

export default BarList