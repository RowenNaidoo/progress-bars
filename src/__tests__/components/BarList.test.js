import React from 'react'
import { shallow } from 'enzyme'
import BarList from '../../components/BarList'


describe('#BarList', () => {
  const component = shallow(<BarList bars={[39, 37]} />)

  test('renders learn react link', () => {
    expect(component).toMatchSnapshot()
  })

  test('renders corresponding number of bars', () => {
    expect(component.find('Bar')).toHaveLength(2)
  })

})