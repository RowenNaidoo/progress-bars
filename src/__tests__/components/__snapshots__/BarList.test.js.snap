// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#BarList renders learn react link 1`] = `ShallowWrapper {}`;
